data/external/scrapers/nat.py

```diff
@@ -42,9 +39,43 @@ def scrape_rooms():
         key = room["room_code"]  # needed, as room_code is removed in _sanitise_room
         rooms[key] = _sanitise_room(room)
 
+    # split the orgs into a different file
+    _extract_orgs(rooms)
+    for room in rooms.values():
+       org_ids = [room.pop("org")["org_id"]]
+       org_ids.extend(org["org"]["org_id"] for org in room.pop("orgs"))
+       room["org_ids"] = org_ids
     return rooms
 
 
+@cached_json("orgs_nat.json")
+def _extract_orgs(rooms: dict) -> dict:
+    """
+    Extract the organisations from the room information.
+    """
+    org_set: set[str] = set(json.dumps(room["org"], sort_keys=True) for room in rooms.values())
+    for room in rooms.values():
+        org_set.update(json.dumps(org["org"], sort_keys=True) for org in room["orgs"])
+
+    orgs = [json.loads(org) for org in org_set]
+
+    # check if there are any duplicates/ inconsistencies
+    org_ids = [org["org_id"] for org in orgs]
+    if len(set(org_ids)) != len(org_ids):
+        cnt = Counter(org_ids) - Counter(set(org_ids))
+        inconsistent_orgs = {org_id for org_id, _ in cnt.items()}
+        for org in sorted(orgs, key=lambda org: org["org_id"]):
+            if org["org_id"] in inconsistent_orgs:
+                logging.warning(f"Inconsistent org: {org}")
+        raise ValueError(f"{len(cnt.items())} Inconsistent orgs")
```

Comment:
Is it realistic that this happens (didn't try it out)? It's not bad to have it there, but reading the code is not straightforward. It might be possible to simplify this whole code by creating a `dict` where the ID is the key and the value a set of JSON strings of all orgs with this id. If any set has more than one value it's an error:

```python
orgs_list = []
for room in rooms.values():
    orgs_list.append(room["org"])
    orgs_list.extend(org["org"] for org in room["orgs"])

orgs = {}
for org in orgs_list:
    # Converting it to a JSON string makes it hashable
    orgs.setdefault(org["org_id"], set()).add(json.dumps(org, sort_keys=True))

for org_id, org_set in orgs.items():
    if len(org_set) > 1:
        raise ValueError(f"Conflicting data for org_id {org_id}:\n", \n.join(json.loads(org) for org in org_set))

    org = org_set.pop()
    org.pop("org_id")
    org.pop("deleted")
    orgs[org_id] = org
```
(code untested)